### ISSUE TITLE
provide a more dev friendly tool for pre-check before commit

### ISSUE THREAD

**Issue by ben1009** [qdrant/qdrant#3488]
**Is your feature request related to a problem? Please describe.**

provide a more dev friendly tool for check before commit
- like makefile in https://github.com/datafuselabs/databend/blob/main/Makefile
- cargo-make in https://github.com/risingwavelabs/risingwave/blob/main/Makefile.toml. 
- instead of manually run `cargo +nightly fmt --all`, `cargo clippy --all --all-features`, desides run `cargo nextest run --workspace --profile ci` one by one

another issue:
if need `nightly`, better install first than report error when commit 
`Found configured hook: pre-push
Running command: cargo +nightly fmt --all -- --check
error: toolchain 'nightly-x86_64-unknown-linux-gnu' is not installed
Configured hook command failed
pre-push hook rejected`

**Comment by timvisee** [comment]
I agree that this could be better. I don't like using makefiles for this though, we can _finally_ avoid these with Rust/cargo.

What about creating two or three scripts such as `tools/reformat.sh` and `tools/clippy-all.sh`?

> if need nightly, better install first than report error when commit

The pre-push should not automatically install toolchains for you. It'll be up to the user to do this, because depending on his system, there may be different ways of doing so.

**Comment by ben1009** [comment]
> I agree that this could be better. I don't like using makefiles for this though, we can _finally_ avoid these with Rust/cargo.
> 
> 
> 
> What about creating two or three scripts such as `tools/reformat.sh` and `tools/clippy-all.sh`?
> 
> 
> 
> > if need nightly, better install first than report error when commit
> 
> 
> 
> The pre-push should not automatically install toolchains for you. It'll be up to the user to do this, because depending on his system, there may be different ways of doing so.

OK,for the first part, I could commit a pr later related to the tools.
